lib.php

```diff
@@ -76,11 +76,18 @@ function local_login_backgroundimage() {
  *
  */
 function local_login_after_config() {
+
     global $CFG, $FULLME;
+    $forcelogin = get_config('local_login', 'forcelogin');
     $noredirect  = optional_param('noredirect', 0, PARAM_BOOL); // Don't redirect.
     if (stripos($FULLME, $CFG->wwwroot.'/login/index.php') === 0 &&
        !isloggedin() && !empty($noredirect) && !empty($CFG->alternateloginurl)) {
-
         unset($CFG->alternateloginurl);
     }
+
+    if ((stripos($FULLME, $CFG->wwwroot.'/index.php') === 0 || (strlen($FULLME) == strlen($CFG->wwwroot."/"))) &&
```

Comment:
I almost merged this as-is, but as the default value in the settings.php page needs to change I wonder if a better solution here might be to use parse_url 

eg something like:
```
$path = parse_url($FULLME, PHP_URL_PATH)
if (($path == "/index.php" || $path == "/") && !isloggedin && $forcelogin == 1)  {
```

Reply:
I updated this code to support custom URLs (/moodle/). Please check.